app/models/ability.rb

```diff
@@ -0,0 +1,13 @@
+class Ability
+  include CanCan::Ability
+  def initialize(user)
+    return unless user.present?
+
+    can :manage, Post, author_id: user.id
+    can :manage, Comment, author_id: user.id
+    can :read, :all
+    return unless user.role == 'admin'
+
+    can :manage, :all
+  end
+end
```

Comment:
- [ ] I was able to delete a post that is not my post and that shouldn't happen only the owner of the post or the admin who delete a post so please make sure to fix that

Reply:
- Well resolved 👍🏽